src/app/components/tables/simple/terra-simple-table.component.ts

```diff
@@ -187,12 +244,14 @@ export class TerraSimpleTableComponent<D> implements OnChanges
 
     private updateHeaderCheckboxState():void
     {
-        if(this._selectedRowList.length === 0) // anything selected?
+        let selectedRows:Array<TerraSimpleTableRowInterface<D>> = this.inputRowList.filter((row:TerraSimpleTableRowInterface<D>) => row.selected === true);
+
+        if(selectedRows.length === 0) // anything selected?
         {
             this.uncheckHeaderCheckbox();
         }
-        else if(this._selectedRowList.length > 0 && this.inputRowList.filter(
-                (r:TerraSimpleTableRowInterface<D>):boolean => !r.disabled).length === this._selectedRowList.length) // all selected?
+        else if(selectedRows.length > 0 && this.inputRowList.filter(
+                (r:TerraSimpleTableRowInterface<D>):boolean => !r.disabled).length === selectedRows.length) // all selected?
```

Comment:
`this.inputRowList.filter(r:TerraSimpleTableRowInterface<D>):boolean => !r.disabled)` am besten auch in eine Funktion auslagern. Vllt `private get disabledRowList`

Reply:
Wird doch nur an der einen Stelle hier verwendet

Reply:
Yes, but `his.inputRowList.filter(r:TerraSimpleTableRowInterface<D>):boolean => !r.disabled)` in a if-statement is terrible to read 😄